src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/db/RecallRepository.kt

```diff
@@ -31,4 +34,17 @@ class RecallRepository(
     } catch (e: JpaObjectRetrievalFailureException) {
       throw RecallNotFoundException(recallId)
     }
+
+  override fun findByRecallId(recallId: RecallId): Recall? =
+    findById(recallId.value).orElse(null)
+
+  @Transactional
+  override fun addDocumentToRecall(recallId: RecallId, recallDocument: RecallDocument) {
+    getByRecallId(recallId).let { recall ->
```

Comment:
what happens if we dont find a recall?

Reply:
getRecallById throws a RecallNotFoundException, although it is guarded from that happening in the RecallDocumentService.  Could add an extra unit test to document that though